### ISSUE TITLE
Training code?

### ISSUE THREAD

**Issue by feilong** [pBFSLab/SUGAR#1]
Great work!  Do you have any plans to release the training code soon?  I just read the manuscript and am interested in the implementation of loss functions.
